### PR TITLE
Add gradient clipping

### PR DESCRIPTION
**Pull request by alxmamaev**

### CODE REVIEW

File: catalyst/callbacks/optimizer.py [hunk 1]
```diff
@@ -47,6 +48,7 @@ def __init__(
         model_key: str = None,
         optimizer_key: str = None,
         accumulation_steps: int = 1,
+        grad_clip_params: dict = None,
```
Comment (Scitator) [inline catalyst/callbacks/optimizer.py]: could you please add `grad_clip_fn: Callable` also?

Comment (Scitator) [inline catalyst/callbacks/optimizer.py]:
```suggestion
        grad_clip_params: Dict = None,
```

File: catalyst/callbacks/optimizer.py [hunk 2]
```diff
@@ -56,6 +58,8 @@ def __init__(
         self.model = None
         self.optimizer = None
         self.criterion = None
+        self.grad_clip_fn = None
```
Comment (Scitator) [inline catalyst/callbacks/optimizer.py]:
```suggestion
        self.grad_clip_fn = grad_clip_fn
```

File: catalyst/callbacks/optimizer.py (hunk 1 above)
Comment (Scitator) [reply catalyst/callbacks/optimizer.py]: btw, could you please add the docs for `grad_clip_fn` and `grad_clip_params`?

Comment (Scitator) [reply catalyst/callbacks/optimizer.py]: and tests :)
they could be placed under `catalyst/callbacks/tests` folder

File: catalyst/callbacks/optimizer.py [hunk 3]
```diff
@@ -72,6 +76,11 @@ def __init__(
             self._prefix_lr = "lr"
             self._prefix_momentum = "momentum"
 
+        if grad_clip_params is not None:
+            self.grad_clip_fn = REGISTRY.get(grad_clip_params.pop("fn_name"))
```
Comment (Scitator) [inline catalyst/callbacks/optimizer.py]: let's use `self.grad_clip_fn = partial(..., **grad_clip_params)`

File: catalyst/callbacks/optimizer.py [hunk 4]
```diff
@@ -94,6 +103,9 @@ def on_batch_end(self, runner: "IRunner"):
             loss = runner.batch_metrics[self.metric_key]
             runner.engine.backward_loss(loss, self.model, self.optimizer)
 
+            if self.grad_clip_fn is not None:
+                self.grad_clip_fn(self.model.parameters(), **self.grad_clip_params)
```
Comment (Scitator) [inline catalyst/callbacks/optimizer.py]:
```suggestion
                self.grad_clip_fn(self.model.parameters())
```

**Review by Scitator — DISMISSED** [review]

**Comment by Scitator** [comment]
btw, could you please also add a note to the `Changelog.md`?

File: catalyst/callbacks/optimizer.py (hunk 1 above)
Comment (Scitator) [inline catalyst/callbacks/optimizer.py]: let's define `grad_clip_fn` during `init` as well - it's a much easier way for Python API users

File: examples/mnist_stages/config.yml [hunk 5]
```diff
@@ -98,6 +98,9 @@ stages:
       optimizer:
         _target_: OptimizerCallback
         metric_key: *model_loss
+        grad_clip_fn: clip_grad_value_
```
Comment (Scitator) [inline examples/mnist_stages/config.yml:101]: could you please also add a test for `catalyst/tests/teste_finetune` or something similar?

File: catalyst/callbacks/optimizer.py [hunk 6]
```diff
@@ -39,6 +41,8 @@ class OptimizerCallback(IOptimizerCallback):
         optimizer_key: a key to select a optimizer from ``runner.optimizer`` in case
             there are several of them and they are in a dictionary format.
         accumulation_steps: number of steps before ``model.zero_grad()``
+        grad_clip_fn: callable gradient cliping function or name or name
```
Comment (Scitator) [inline catalyst/callbacks/optimizer.py]:
```suggestion
        grad_clip_fn: callable gradient cliping function or it's name or
```
